### DISCUSSION TITLE
TIFF alpha channel regression between 7.1.1.40 and 7.1.1.41

### DISCUSSION THREAD

**Discussion by bohemima** [ImageMagick/ImageMagick#7848] [Development] [answered]
### ImageMagick version

7.1.1.41

### Operating system, version and so on

RockyLinux 9

### Description

The following command produced a png with transparent background using v7.1.1.40 but does not properly do the same with 7.1.1.41, it produces a png with a black background.

```convert -alpha on -channel rgba ./original.tif[0] -alpha set -intent Perceptual -black-point-compensation -profile sRGB_IEC61966-2-1.icc -auto-orient -resize 2000x2000 -background transparent -strip +write test.png```

Unfortunately I'm unable to share the original TIFF file but it should be reproducible with any TIFF including a alpha channel, not sure if the size of the TIFF might be related, all our test files are quite big.

**Comment by fmw42** [comment]
Please post your input TIFF file.  On IM 7, use magick not convert.  Your syntax is faulty and especially for IM 7.  Read the input first right after magick. Also see my other changes. +write is not necessary unless you add null: at the end.

Try

`magick ./original.tif[0] -alpha on -channel rgba -intent Perceptual -black-point-compensation -profile sRGB_IEC61966-2-1.icc -auto-orient -background transparent -resize 2000x2000 -strip test.png`

**Reply by bohemima** [reply]
Hi, 

Thanks, the `magick` has the same issue with 7.1.1.41 but works in 7.1.1.40. The `convert` command we use is a part of a much larger command but simplified to show the error.

Unfortunately I'm unable to share the TIFF publicly, it happened for 4 of our customers on a wide range if TIFF files since upgrading to 7.1.1.41. If possible I could maybe share an example file privately after I check with the customer.

**Reply by dlemstra** [reply]
We really need an image to properly test this. If you cannot find one freely available that can be used to reproduce this then feel free to reach to me by email (see my profile) so we can discuss how we can get access to a file that reproduces this issue. It must be something related to your image because this seems to work:

```console
> magick logo: -fill none -opaque white output.tiff

> magick identify -format %[channels] output.tiff
srgba 4.0
```

**Reply by bohemima** [reply]
Hi,

Found an image you can test on, attaching my observations as well. https://drive.google.com/file/d/1wpb2zoSnQ1LhTA82sFxF651STgL7j6Hw/view?usp=sharing

```
sh-5.1$ magick -version
Version: ImageMagick 7.1.1-41 Q16-HDRI x86_64 22504 https://imagemagick.org
Copyright: (C) 1999 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Features: Cipher DPC HDRI Modules OpenMP(4.5)
Delegates (built-in): bzlib cairo djvu fftw fontconfig freetype gslib gvc heic jbig jng jp2 jpeg jxl lcms lqr ltdl lzma openexr pangocairo png ps raqm raw rsvg tiff webp wmf x xml zip zlib zstd
Compiler: gcc (11.4)
sh-5.1$ magick CarTransparentBackground.tif[0] -alpha on -channel rgba -auto-orient -background transparent -resize 2000x2000 -strip test.png
sh-5.1$ magick identify -format %[channels] test.png
srgba 4.0
sh-5.1$ magick identify -verbose test.png  | grep -i alpha
  Type: TrueColorAlpha
    Alpha: 1-bit
    Alpha:
sh-5.1$ magick identify -format %[channels] test.png
srgba 4.0
```

Even though `magick identify -format %[channels] test.png` returns srgba 4.0, note that `identify verbose` shows `Alpha: 1-bit`  and when looking at the PNG there is no transparency.

Downgrading to 7.1.1.40 yields different result, namely `Alpha: 8-bit` and the PNG does have transparency

```sh-5.1$ magick -version
Version: ImageMagick 7.1.1-40 Q16-HDRI x86_64 22476 https://imagemagick.org
Copyright: (C) 1999 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Features: Cipher DPC HDRI Modules OpenMP(4.5)
Delegates (built-in): bzlib cairo djvu fftw fontconfig freetype gslib gvc heic jbig jng jp2 jpeg jxl lcms lqr ltdl lzma openexr pangocairo png ps raqm raw rsvg tiff webp wmf x xml zip zlib zstd
Compiler: gcc (11.4)
sh-5.1$ magick CarTransparentBackground.tif[0] -alpha on -channel rgba -auto-orient -background transparent -resize 2000x2000 -strip test.png
sh-5.1$ magick identify -verbose test.png  | grep -i alpha
  Type: TrueColorAlpha
    Alpha: 8-bit
    Alpha:
  Alpha: none   #00000000
sh-5.1$ magick identify -format %[channels] test.png
srgba 4.0```

Let me know if you need me to test anything else.

**Reply by dlemstra** [reply]
Attaching the file here for future reference: [CarTransparentBackground.zip](https://github.com/user-attachments/files/18218521/CarTransparentBackground.zip)

**Reply by dlemstra** [reply, marked as answer]
Thanks for providing that file. It looks like a refactor of mine where I removed some duplicate code broke things. I just pushed a patch to fix this because it looks like we need to read the sample info right before we use it. This will be resolved in the next release.

**Reply by bohemima** [reply]
Happy to help, thanks!

**Reply by fmw42** [reply]
This command works fine for me on IM 7.1.1.41 Mac OSX Ventura

`magick ./original.tif[0] -alpha on -channel rgba -intent Perceptual -black-point-compensation -profile sRGB.icc -auto-orient -background transparent -resize 2000x2000 -strip test.png`

![test](https://github.com/user-attachments/assets/566639f9-9572-440f-82f3-def73965b88e)
